### DISCUSSION TITLE
Concurrent Users - Performance?

### DISCUSSION THREAD

**Discussion by KoalaGeo** [protomaps/PMTiles#433] [Q&A] [answered]
Hi,

Do you have any examples / info on performance with concurrent users? 

If I've a visualisation which I know gets over concurrent 500 users at peak times how does a pmtiles handle that? 

Or is there an expectation to run replication with multiple copies & load balancing? 

Or am I missing something about how pmtiles works?

Regards, 

@Koalageo

**Comment by bdon** [comment, marked as answer]
Depends on where you're storing it; if you are using a static storage backend like S3 or a popular fileserver with HTTP Range Requests like NGINX, 500 concurrent users should not be an issue.

I'm not aware of a situation where replication with multiple copies and load balancing is necessary for throughput, but multiple regional copies can help with latency.

**Reply by KoalaGeo** [reply]
Sorry and clarifying the docs, of I'm hosting on standard nginx docker image can I just copy the pmtiles file in, or do I need to run pmtiles cli 'pmtiles serve .' to actually have it working?

**Reply by bdon** [reply]
plain nginx = you need to use the pmtiles javascript client to decode in the browser
`pmtiles serve` = will serve z/x/y tiles, you don't need the pmtiles javascript client

**Comment by KoalaGeo** [comment]
Happy days! Thanks for the quick reply!
